version: '3'

networks:
  scan-api:
    external: false

services:
  scan-web-api:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: scan-web-api
    ports:
      - '3031:3031'
    networks:
      - scan-api
    depends_on:
      - postgres
    restart: on-failure
    environment:
      - PORT=3031
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DATABASE=${POSTGRES_DATABASE}
      - POSTGRES_HOST=host.docker.internal
      - POSTGRES_PORT=5432

  postgres:
    image: postgres:14.2-alpine
    container_name: scan-database
    networks:
      - scan-api
    env_file:
      - .env
    environment:
      - TZ=Europe/Moscow
      - PGTZ=Europe/Moscow
    ports:
      - '5432:5432'
    volumes:
      - scan-pgdata:/var/lib/postgresql/data
      - /etc/localtime:/etc/localtime:ro

volumes:
  scan-pgdata:
    external: false
